text/0000-create-stackify-repo.md

```diff
@@ -0,0 +1,141 @@
+# Meta
+[meta]: #meta
+- Name: Create stackify repo
+- Start Date: 2020-11-06
+- Authors: @martyspiewak @mdelillo @dumez-k
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+Create a new repo called “stackify” which can be used to turn OCI images into CNB compliant stack images.
+
+# Definitions
+[definitions]: #definitions
+N/A
+
+# Motivation
+[motivation]: #motivation
+Currently, for users to extend the published stack images, or turn their own images into stack images, they have to create Dockerfiles which take the base images, apply their changes, and then set the CNB metadata. It can be difficult to generate and set all of the relevant metadata, especially the mixins.
+
+Stackify will be a tool that they can use to perform all of these manipulations to their base image without Dockerfiles.
+
+
+# What it is
+[what-it-is]: #what-it-is
+Stackify will be a tool that will be useful to users who would like to customize the published stack images or turn their own base images into CNB compliant stack images.
+
+Users can use stackify to add packages and ca-certificates to a base image as well as add all required CNB metadata to the image.
+
+Stackify can be integrated into Pack in the form of a `pack create-stack` command.
+
+# How it Works
+[how-it-works]: #how-it-works
+For stacks defined by the buildpacks project, Stackify will add mixins labels for each package installed on the image. If Stackify is used to give add a stack ID other than one defined by the buildpacks project, mixin labels will not be added.
+
+If Stackify is used with an already valid stack image, the `CNB_*` labels and users will not be changed unless new user and group ID's are passed, in which case the labels and user will be overwritten.
+
+If run on the image being "stackify-ed", Stackify can be used to add packages and ca-certificates.
+
+Stackify will use Kaniko to create additional image layers with the additional packages and ca-certificates that users choose to add to their base image. It will also use GGCR to manipulate the image metadata.
+
+
+### Proposed Interface
+Stackify will be run once for each image (build and run):
+```
+stackify 
+  --base-image string
+    	Source image ref
+  --stack-image string
+    	Destination for stack image
+  --certs-dir string
+    	Path to directory with CA Certs to add to image
+  --group-id string
+    	CNB Group ID
+  --image-type string
+    	Type of image to create (build or run)
+  --mixins string
+    	Comma-separated list of mixins to add to labels
+  --packages string
+    	Comma-separated list of packages to add to image
+  --stack-id string
+      Stack ID to add to image metadata
+  --user-id string
+    	CNB User ID
+```
+
+This will be called under the hood by the `pack create-stack` command, which will take a TOML config file:
+
+`pack create-stack my-stack -c /path/to/config.toml`
+
+Config file:
+```
+[build]
```

Comment:
This seems like it would be worthwhile to add. It seems like the `released` date should be something stackify can set with the time stackify was called.

I'm not sure what we should do about that second question- seems like something the project as a whole needs to resolve.

Reply:
Maybe I am overthinking the label name issue. Let not worry about that.

Label name aside, maybe it would make the most sense if we could set these values either in a top level `[stack]` table (apply to both build and run images) or set any of these fields besides `id` (which must be the same for both images) in the `[build]` or `[run]` tables, to allow different values to be set on build and run images.